### PR TITLE
feat(app): Add cron jobs to app chart

### PR DESCRIPTION
**Pull request by bsushmith**
Ties up with this guardian issue - https://github.com/odpf/guardian/issues/191 to be able to run guardian jobs separately as cron jobs.

### CODE REVIEW

File: stable/guardian-jobs/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```
Comment (snorlacs) [inline stable/guardian-jobs/Chart.yaml:1]: Lets use v2, so that we enforce helm3 usage

**Comment by rahmatrhd** [comment]
is it possible to have this as part of [stable/guardian](https://github.com/odpf/charts/tree/main/stable/guardian) only?
cc @AkarshSatija @rohilsurana @ravisuhag

**Comment by ravisuhag** [comment]
@bsushmith Yes, let's make this part of the Guardian chart only. cc @rahmatrhd

**Comment by bsushmith** [comment]
@ravisuhag @rahmatrhd made the changes as part of the app chart itself. Felt it could be useful for other applications, so instead of just for guardian, added to app chart itself directly.

**Comment by ravisuhag** [comment]
@rahmatrhd @rohilsurana can you folks please check this?

**Comment by rahmatrhd** [comment]
@bsushmith can you add default values in https://github.com/odpf/charts/blob/main/stable/guardian/values.yaml for running guardian jobs as well?

File: stable/app/templates/cron.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{{- if .Values.cron.enabled }}
+{{- range $job := .Values.cron.jobs }}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: "{{ include "app.fullname" . }}"
```
Comment (rohilsurana) [inline stable/app/templates/cron.yaml:6]: How would this work if we have multiple cron jobs defined?
Wouldn't same name have clashes?

File: stable/app/README.md [hunk 3]
```diff
@@ -40,57 +40,61 @@ The command removes all the Kubernetes components associated with the chart and
...
-|                                                                      |
-| telegraf.containerPort                | int    | `8125`                
-|                                                                      |
-| telegraf.protocol                     | string | `UDP`                
-|                                                                      |
-| telegraf.config                       | string | `""`                | telegraf config file content |
+| cron.enabled           | bool | `false`                                                                                       |  |
+| cron.job.entryPoint    | string | `""`                                                                                          |  |
+| cron.job.name          | string | `""`                                                                                  |  |
+| cron.job.schedule      | string | `"0 0 * * *"`                                                                                 |  |
```
Comment (rohilsurana) [inline stable/app/README.md:59]: Doesn't seem to match the changes made in templates and values.

File: stable/app/templates/cron.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+              command:
+              {{- range $job.entryPoint }}
+                - {{ . }}
+              {{- end }}
+              envFrom:
+                - configMapRef:
+                    name: "{{ include "app.fullname" . }}"
+                - secretRef:
+                    name: "{{ include "app.fullname" . }}-secret"
+          restartPolicy: Never
```
Comment (rohilsurana) [inline stable/app/templates/cron.yaml:33]: Let's add this as a variable, as it could be something widely different as per use-case.

**Comment by bsushmith** [comment]
> @bsushmith can you add default values in https://github.com/odpf/charts/blob/main/stable/guardian/values.yaml for running guardian jobs as well?

@rahmatrhd will add that as a separate pr once this is done since that would involve updating app dependency in the guardian chart.

File: stable/app/templates/cron.yaml (hunk 4 above)
Comment (bsushmith) [reply stable/app/templates/cron.yaml]: done. added it as a separate job level variable.

File: stable/app/templates/cron.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        {{- include "app.selectorLabels" . | nindent 8 }}
+        spec:
+          containers:
+            - name: {{ $job.name }}
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              command:
+              {{- range $job.entryPoint }}
+                - {{ . }}
+              {{- end }}
```
[the remark this answers is not in the document]
Comment (bsushmith) [reply stable/app/templates/cron.yaml]: done. made the changes accordingly.

File: stable/app/README.md (hunk 3 above)
Comment (bsushmith) [reply stable/app/README.md]: fixed and updated the docs

File: stable/app/templates/cron.yaml (hunk 2 above)
Comment (bsushmith) [reply stable/app/templates/cron.yaml]: yes. there would be a name clash.  updated the name value to take in app-name and job-name both concatenated.

**Review by rohilsurana — APPROVED** [review]

**Pull request merged**